python/tests/cli/test_console.py

```diff
@@ -134,14 +134,17 @@ def update_namespace_properties(
 MOCK_ENVIRONMENT = {"PYICEBERG_CATALOG__PRODUCTION__URI": "test://doesnotexist"}
 
 
-def test_missing_uri() -> None:
-    runner = CliRunner()
-    result = runner.invoke(run, ["list"])
-    assert result.exit_code == 1
-    assert (
-        result.output
-        == "URI missing, please provide using --uri, the config or environment variable \nPYICEBERG_CATALOG__DEFAULT__URI\n"
-    )
+def test_missing_uri(empty_home_dir_path: str) -> None:
```

Comment:
This looks good to me! I'd rather use the decorator `@mock.patch.dict(os.environ, MOCK_ENVIRONMENT)` similar as the test below, but I'm aware that we that we can't use the fixture on a global scope.

Reply:
Yeah me too. If you're adamant about it, could do the following, but I'd say it's a pretty terrible workaround 

```
def test_missing_uri(empty_home_dir_path)
    @mock.patch.dict(os.environ, {"HOME": empty_home_dir_path})
    def _test_missing_uri():
        ...
    _test_missing_uri()
```

Reply:
Alternatively, we could do something like this in `conftest.py`:

```
def make_temporary_home_folder(tmp_path_factory) -> None:
    home_path = str(tmp_path_factory.mktemp("home"))
    os.environ["TMP_HOME_PATH"] = home_path
```

and then in the test

```
@mock.patch.dict(os.environ, {"HOME": os.environ["TMP_HOME_PATH"]})
def test_missing_uri():
    ...
```

Reply:
But then we also should invoke the `make_temporary_home_folder`, I prefer the current solution.